docs/configuration.rst

```diff
@@ -0,0 +1,242 @@
+        <domain_participant name="MyPubParticipant" domain_ref="MyDomainLibrary::MyDomain">
+            <publisher name="MyPublisher">
+                <data_writer name="MySquareWriter" topic_ref="Square" />
+            </publisher>
+        </domain_participant>
+
+        <domain_participant name="MySubParticipant" domain_ref="MyDomainLibrary::MyDomain">
+            <subscriber name="MySubscriber">
+                <data_reader name="MySquareReader" topic_ref="Square" />
+            </subscriber>
+        </domain_participant>
+    </domain_participant_library>
+
+For more information about the format of a participant library, see
+`XML-Based Application Creation: Participant Library <https://community.rti.com/static/documentation/connext-dds/current/doc/manuals/connext_dds/xml_application_creation/html_files/RTI_ConnextDDS_CoreLibraries_XML_AppCreation_GettingStarted/index.htm>`__
+
+Quality of service
+~~~~~~~~~~~~~~~~~~
+
+All DDS entities have an associated `quality of service (Qos) <https://community.rti.com/static/documentation/connext-dds/current/doc/manuals/connext_dds/html_files/RTI_ConnextDDS_CoreLibraries_UsersManual/index.htm#UsersManual/QosPolicies.htm>`__.
+There are several ways to configure it.
+
+You can define a Qos profile and make it the default. The following example
+configures all data readers and data writers with reliable and transient-local Qos:
+
+.. code-block:: xml
+
+    <qos_library name="MyQosLibrary">
+        <qos_profile name="MyQosProfile" is_default_qos="true">
+            <datareader_qos>
+                <reliability>
+                    <kind>RELIABLE_RELIABILITY_QOS</kind>
+                </reliability>
+                <durability>
+                    <kind>TRANSIENT_LOCAL_DURABILITY_QOS</kind>
+                </durability>
+            </datareader_qos>
+            <datawriter_qos>
+                <reliability>
+                    <kind>RELIABLE_RELIABILITY_QOS</kind>
+                </reliability>
+                <durability>
+                    <kind>TRANSIENT_LOCAL_DURABILITY_QOS</kind>
+                </durability>
+            </datawriter_qos>
+        </qos_profile>
+        ...
+    </qos_library>
+
+You can define the Qos for each individual entity:
+
+.. code-block:: xml
+
+    <domain_participant name="MyPubParticipant" domain_ref="MyDomainLibrary::MyDomain">
+        <participant_qos> ... </participant_qos>
+        <publisher name="MyPublisher">
+            <publisher_qos> ... </publisher_qos>
+            <data_writer name="MySquareWriter" topic_ref="Square">
+                <datawriter_qos>
+                    <reliability>
+                        <kind>RELIABLE_RELIABILITY_QOS</kind>
+                    </reliability>
+                    <durability>
+                        <kind>TRANSIENT_LOCAL_DURABILITY_QOS</kind>
+                    </durability>
+                </datawriter_qos>
+            </data_writer>
+        </publisher>
+        ...
```

Comment:
delete ...

Reply:
In general, I think we should provide at least valid XML as the snippets (e.g., in this snippet remove the <participant_qos> tags - could add an XML comment if we want to convey that participant_qos goes there